### PR TITLE
Replace Redux with MobX

### PR DESCRIPTION
**Pull request by ericnishio**

### CODE REVIEW

**Comment by hugovk** [comment]
The CI failed:
```
Error: src/types.js:3
  3: import type {RootState as $RootState} from './root/types'
                                                ^^^^^^^^^^^^^^ ./root/types. Required module not found
Error: src/types.js:4
  4: import type {Selector as ReselectSelector} from 'reselect'
                                                     ^^^^^^^^^^ reselect. Required module not found
```

**Comment by cniska** [comment]
@ericnishio Whoa, that was quick. Thanks for this.

**Comment by ericnishio** [comment]
@crisu83 That's because 94.5% of the work was done by mashing the delete key :)

File: src/hello/store.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// @flow
+
+import {observable, action} from 'mobx'
+
+class HelloStore {
+  @observable message: string = 'Hello.'
+
+  @action
```
Comment (ericnishio) [inline src/hello/store.js:8]: TBH, I'd prefer to have Prettier place the `@action` decorator on the same line as the method.

Comment (cniska) [reply src/hello/store.js:8]: @ericnishio Can't we change the formatting to allow that?

**Comment by cniska** [comment]
Remember to add the (fixes #63) to one of the commits, so that the related issue is automatically closed. The best way to do this would be to do an interactive rebase and rephrase one of the commit messages.

File: src/hello/store.js (hunk 1 above)
Comment (ericnishio) [reply src/hello/store.js:8]: @crisu83 Yea. I just haven't found the right ESLint rule for it.

I don't mind having the `@observer` decorator above the `class` keyword. In fact, I prefer it that way. But `@action` (for methods), `@observable` (for properties), and `@computed` (for getters) should be placed on the same line.

It's only a matter of taste so some might disagree :)

File: package.json [hunk 2]
```diff
@@ -45,6 +42,7 @@
     "whatwg-fetch": "2.0.3"
   },
   "devDependencies": {
+    "babel-plugin-transform-decorators-legacy": "^1.3.4",
```
Comment (cniska) [inline package.json:45]: This is probably required to compile the project, therefore it should be under `dependencies`.

**Review by cniska — APPROVED** [review]
Looks good to me, I think we can merge this and improve it as we go.

File: package.json (hunk 2 above)
Comment (cniska) [reply package.json:45]: I fixed this already.

**Pull request merged**